nbs/15_gsa_gls/07-compile_gwas_snps_and_twas_genes.ipynb

```diff
@@ -0,0 +1,2355 @@
+   },
+   "source": [
+    "# GWAS"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "331e5691-c014-43e4-b94c-060c64d329df",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gwas_file_columns = pd.read_csv(GWAS_FILE, sep=\"\\t\", nrows=2).columns\n",
+    "\n",
+    "assert (\n",
+    "    \"panel_variant_id\" in gwas_file_columns\n",
+    "), \"GWAS file must be the final imputed one with column 'panel_variant_id'\"\n",
+    "\n",
+    "assert (\n",
+    "    \"pvalue\" in gwas_file_columns\n",
+    "), \"GWAS file must be the final imputed one with column 'pvalue'\"\n",
+    "\n",
+    "assert (\n",
+    "    \"zscore\" in gwas_file_columns\n",
+    "), \"GWAS file must be the final imputed one with column 'zscore'\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8d419629-8f4d-4c83-bed3-cfc071a58090",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gwas_data = pd.read_csv(\n",
+    "    GWAS_FILE,\n",
+    "    sep=\"\\t\",\n",
+    "    usecols=[\"panel_variant_id\", \"pvalue\", \"zscore\"],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3f4e814e-6b41-407e-9099-86795eec6ed9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gwas_data.shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b4a906e9-c80a-47fc-9d73-5adb046f8b76",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gwas_data.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "033c24fb-9c60-4ffd-929e-b0e6089820ae",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gwas_data.dropna().shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "28232ce5-0afd-464e-9be3-64c3a68c2b25",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# remove SNPs with no results\n",
+    "gwas_data = gwas_data.dropna()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d9a6c31f-eedd-43e5-b762-5a9663be1b77",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gwas_data.shape"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8da344de-2aa5-48a9-98ec-e8c3397ae7e6",
+   "metadata": {
+    "papermill": {
+     "duration": 0.016403,
+     "end_time": "2021-04-28T22:27:17.896463",
+     "exception": false,
+     "start_time": "2021-04-28T22:27:17.880060",
+     "status": "completed"
+    },
+    "tags": []
+   },
+   "source": [
+    "## Save GWAS variants"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ac11fb33-ea3c-4d5c-a90b-0ceebdac43d3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gwas_data.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5f34bb3d-761a-4860-a179-77ea5f5784ad",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# in eMERGE, some values here are repeated\n",
```

Comment:
Do you do anything different further down based on if there are repeat values, or is it just for the viewer's information?

Reply:
It's just for the viewer's information. Actually, those values are removed later. I updated the description.